plugins/modules/files/json_file.py

```diff
@@ -0,0 +1,538 @@
+#    }
+# }
+
+- name: "Ensure values are present in the JSON diff, check only on values."
+  community.general.json_file:
+    path: "/my/dest/file.json"
+    indent: 2
+    value:
+      A: 1
+    state: "present"
+# We will change the indent but when the change are only detected on the values
+# the JSON file will not be update ("A": 1 is already in the originals values).
+# There is no change.
+# Get the following content in /my/dest/file.json :
+# {
+#    "A": 1,
+#    "B": 3,
+#    "C": {
+#        "D": 3,
+#        "E": [
+#            4,
+#            5,
+#            6
+#        ],
+#        "F": 7
+#    }
+# }
+
+- name: "Ensure values are present in the JSON diff, also check file format."
+  community.general.json_file:
+    path: "/my/dest/file.json"
+    indent: 2
+    diff_on_value: false
+    value:
+      A: 1
+    state: "present"
+# There is not diff in the values but the file format is changed (intent 4
+# There is a change.
+# Get the following content in /my/dest/file.json :
+#{
+#  "A": 1,
+#  "B": 2,
+#  "C": {
+#    "D": 3,
+#    "E": [
+#      4,
+#      5,
+#      6
+#    ]
+#  }
+#}
+'''
+
+RETURN = r'''
+  changed:
+    description:
+      - Indicate whether the JSON file is changed or not.
+    returned: always
+    type: bool
+  created:
+    description:
+      - Indicate if the file JSON is created or not.
+    returned: success
+    type: bool
+  diff:
+    description:
+      - Contain a dict with two elements, C(after) contain the value after
+        the change and C(before) contain values before the change.
+    returned: changed
+    type: dict
+  failed:
+    description:
+      - Indicate if the module has failed or not.
+    returned: always
+    type: bool
+  result:
+    description:
+      - Contain the values that are finally present in the JSON file.
+    returned: success
+    type: dict
+'''
+
+import os
+import json
+from ansible_collections.community.general.plugins.module_utils.data_merge_utils import DataMergeUtils
+from ansible_collections.community.general.plugins.module_utils.mh.module_helper_dest_file import (
+    DestFileModuleHelper,
+    ModuleHelperException,
+)
+from ansible_collections.community.general.plugins.module_utils.common.validation import check_type_dict_or_list
+
+
+class JsonFile(DestFileModuleHelper):
+
+    module = dict(
+        argument_spec=dict(
+            path=dict(type='path', required=True, aliases=['dest']),
+            state=dict(
+                type='str',
+                default='present',
+                choices=[
+                        'present',
+                        'absent',
+                        'identic',
+                ]
+            ),
+            list_diff_type=dict(
+                type='str',
+                default='value',
+                choices=[
+                        'value',
+                        'index',
+                ],
+            ),
+            value=dict(type=check_type_dict_or_list, required=True),
```

Comment:
Use `type=raw` and handle type validation in the code:
```suggestion
            value=dict(type='raw', required=True),
```